### PR TITLE
Feature/feature refactor library

### PR DESCRIPTION
**Pull request by enrichp**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrUpdateWash.java  -16
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrObtainAllPetWashes.java  -11
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrDeleteWash.java  -14
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrUpdateWash.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrObtainAllPetWashes.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrDeleteWash.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5498439&bid=18021178)

**Review by AlbertPG — DISMISSED** [review]
Okey, I see no problem

File: app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Meals.java [hunk 1]
```diff
@@ -18,9 +18,9 @@ public Meals(DateTime dateTime, Double kcal, String mealName) {
     }
 
     public Meals(Meal meal) {
-        super(MEAL + meal.getBody().getMealName() + OF_THE_DAY + meal.getDate(),
-            DateTime.Builder.buildFullString(meal.getDate()));
-        this.mealDate = DateTime.Builder.buildFullString(meal.getDate());
+        super(MEAL + meal.getBody().getMealName() + OF_THE_DAY + meal.getBody(),
+            DateTime.Builder.buildFullString(meal.getBody().toString()));
+        this.mealDate = DateTime.Builder.buildFullString(meal.getBody().toString());
```
Comment (xavier-campos-diaz) [inline app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Meals.java]: No entenc perquè canvies el meal.getDate() per un meal.getBody

**Review by xavier-campos-diaz — COMMENTED** [review]
He deixat un comentari perquè hi ha un canvi que no entenc

File: app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Meals.java (hunk 1 above)
Comment (enrichp) [reply app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Meals.java]: Ahora ya no existe el get Date el Marc lo ha cambiado era getKey no getBody me había equivocado

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrUpdateWash.java  -16
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrDeleteWash.java  -14
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrObtainAllPetWashes.java  -11
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrUpdateWash.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrDeleteWash.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/washes/TrObtainAllPetWashes.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5498439&bid=18021178)

**Review by danicm47 — APPROVED** [review]

**Review by AlbertPG — APPROVED** [review]
Okey, if you have coorected the issue indicated by Xavi, I'll approve it

**Review by xavier-campos-diaz — APPROVED** [review]

**Pull request merged**